### ISSUE TITLE
Fable typeof seems to be broken in versions > 3.7

### ISSUE THREAD

**Issue by zelenij** [fable-compiler/repl#152]
This code

```
open System
open Fable.Core
open Fable.Core.JsInterop

[<Emit("window.alert($0)")>]
let alert (message: string): unit = jsNative

type Test = {
    Name: string
}

type R = 
    | A of int
    | B of string

type Service<'a> = {
    GetRecord: int -> int64 -> Async<'a>
}
with
    static member inline RouteBuilder _ m =
        let name = typeof<'a>.Name
        sprintf "/api/%s/%s" name m

let a = R.A 72
let oa = a.GetType().Name
let a1 = Service<R>.RouteBuilder "" "Add"
alert(a1)
```

when run in the online tool with fable 3.7.9 produces

```
/api/String/Add
```

Expected:

```
/api/R/Add
```

And with version 3.7.0 it does that.

**Comment by alfonsogarciacaro** [comment]
This is a tricky thing, maybe we should report it to the F# compiler. The problem is that the F# compiler assigns the same name to the generic automatically inferred for the unused argument and this confuses Fable. If you explicitly type the generic argument the problem is solved.

```fsharp
static member inline RouteBuilder (_: 'b) m =
```

Unfortunately I'm afraid I cannot easily fix this in Fable but I will add a warning to the next version so users can fix the issue in their code. Thanks for reporting!

![image](https://user-images.githubusercontent.com/8275461/165700841-c0e12aed-3522-4f71-8b7b-28713a9d1b60.png)

**Comment by zelenij** [comment]
I see, interesting...

Any idea why it actually works in Fable 3.7.0 and below?

**Comment by inosik** [comment]
This might be a regression in F#. In VS 2019 the signature of `RouteBuilder` is `'a0 -> string ->string` when hovering over it.

**Comment by alfonsogarciacaro** [comment]
Actually I found a work-around, see last part [of this comment](https://github.com/dotnet/fsharp/issues/13062#issuecomment-1114094230). If you upgrade to Fable 3.7.11 it should work without the explicit annotations.

**Comment by zelenij** [comment]
Nice one! 👍

**Comment by ncave** [comment]
@alfonsogarciacaro Do we want to revert this work-around, now that [dotnet/fsharp#13062](https://github.com/dotnet/fsharp/issues/13062) is supposedly fixed?

**Comment by alfonsogarciacaro** [comment]
@ncave Unfortunately this still seems to be a problem. With latest Fable version (which should contain latest FCS bits) in this code:

```fsharp
type MyRecord<'a> =
    { Value: 'a }
    // F# automatically still assigns 'a name to argument's generic type

    static member Stringify v = string v
```

The `Stringify` method gets compiled in Dart as follows, which causes an error because the name of the generics is the same:

```dart
String MyRecord$1_Stringify_2B595<$a, $a>($a v) => v.toString();
```

**Comment by ncave** [comment]
@alfonsogarciacaro That's ok, for Rust it was the other way (generating an unnecessary type), so I removed the work-around for Rust only.
